### ISSUE TITLE
Storage Drawers should be water-loggable for underwater awesomeness

### ISSUE THREAD

**Issue by jsuereth** [jaquadro/StorageDrawers#826]
Would be awesome if storage drawers could be waterlogged to allow underwater storage centers in 1.15+.

I have a patch to 1.15 that appears to make this work, but has some clipping issues in various viewing angles.    Wanted to first check with you on whether or not you'd be amenable to such a patch before spending more time on it.

Screenshots:

Non-waterlogged storage drawer ![previous](https://i.imgur.com/noBTkRB.png)
Waterlogged storage drawer (uses my patch): ![waterlogged](https://i.imgur.com/PHbZ9fm.png)
The current rendering issue I'd need to look into ![clipped render](https://imgur.com/g51dpt9.png)

TL;DR: on the work I've discovered (for 1.15 branch):

1. Register `BlockStateProperties.WATERLOGGED` as a property of storage drawers
2. Implement `IWaterLoggable`
3. Override `getFluidState` to return still water state if waterlogged.
4. Override `getStateForPlacement` to check if placing into a block with FluidState of an appropriate water level, if so auto-waterlog the block.

Not sure where to look for the rendering issue that clips the front, but I can try to chase that down.

**Comment by eSonOfAnder** [comment]
This would be a great fix.

Regarding the rendering issue in the last image, that's (most likely) not related to your waterlogged change - that's been reported multiple times here and is related to there being a solid block directly in front of the half drawer.
